src/macaron/repo_finder/provenance_finder.py

```diff
@@ -35,191 +37,350 @@ def __init__(self) -> None:
                 elif isinstance(registry, JFrogMavenRegistry):
                     self.jfrog_registry = registry
 
-    def find_provenance(self, purl: PackageURL) -> InTotoPayload | None:
-        """Find the provenance files of the passed PURL.
+    def find_provenance(
```

Comment:
To me, this method looks like it only returns either:
- An empty list
- An instance of IntotoPayload (from my understanding, we go through the list of `discovery_functions` and return the very first IntotoPayload that we find).
Therefore, I wonder why we need the return value of this method to be `list[InTotoPayload]`?

Reply:
Ah nevermind, it looks like the list of InTotoPayload is used here because we decide that each discovery function must returns a list of InTotoPayload instances.

Reply:
Yes, and they return a list because the pre-existing npm function has to return two provenances. In most other cases I expect it will be a single item list.